src/server/locals.js

```diff
@@ -0,0 +1,161 @@
+/* globals Map */
+var inspector = require('inspector');
+var async = require('async');
+
+function Locals(config) {
+  this.config = config;
+  this.errorCounter = 0;
+  this.currentErrors = new Map();
+
+  this.initSession();
+}
+
+Locals.prototype.initSession = function() {
+  this.session = new inspector.Session();
+  this.session.connect();
+
+  this.session.on('Debugger.paused', ({ params }) => {
+    if (params.reason == 'promiseRejection' || params.reason == 'exception') {
+      var key = params.data.description;
+      this.currentErrors.set(key, params);
+
+      if (this.currentErrors.size > 4) {
+        var firstKey = this.currentErrors.keys()[0];
+        this.currentErrors.delete(firstKey);
+      }
+    }
+  });
+
+  this.session.post('Debugger.enable', (_err, _result) => {
+    this.session.post('Debugger.setPauseOnExceptions', { state: 'all'}, (_err, _result) => {
+    });
+  });
+}
+
+Locals.prototype.currentLocalsMap = function() {
+  if (this.currentErrors.size) {
+    return new Map(this.currentErrors);
+  }
+}
+
+Locals.prototype.mergeLocals = function(localsMap, stack, key, callback) {
+  var matchedFrames;
+
+  try {
+    var localParams = localsMap.get(key);
+    matchedFrames = this.matchFrames(localParams, stack.reverse());
+  } catch (e) {
+    console.log(e);
+    return callback(e);
+  }
+
+  this.getLocalScopesForFrames(matchedFrames, function(err) {
+    callback(err);
+  });
+}
+
+// Finds frames in localParams that match file and line locations in stack.
+Locals.prototype.matchFrames = function(localParams, stack) {
+  var matchedFrames = [];
+  var localIndex = 0, stackIndex = 0;
+  var stackLength = stack.length;
+  var callFrames = localParams.callFrames;
+  var callFramesLength = callFrames.length;
+
+  for (; stackIndex < stackLength; stackIndex++) {
+    while (localIndex < callFramesLength) {
+      if (this.firstFrame(localIndex, stackIndex) || this.matchedFrame(callFrames[localIndex], stack[stackIndex])) {
+        matchedFrames.push({
+          stackLocation: stack[stackIndex],
+          callFrame: callFrames[localIndex]
+        });
+        localIndex++;
+        break;
+      } else {
+        localIndex++;
+      }
+    }
+  }
+
+  return matchedFrames;
+}
+
+Locals.prototype.firstFrame = function(localIndex, stackIndex) {
+  return !localIndex && !stackIndex;
+}
+
+Locals.prototype.matchedFrame = function(callFrame, stackLocation) {
```

Comment:
same comment about this function as the above for `firstFrame` - this function isn't making use of `this` - does it need to be exported on the `Locals.prototype`?